fe/fe-core/src/main/java/org/apache/doris/plugin/dialect/HttpDialectConverterPlugin.java

```diff
@@ -100,10 +123,43 @@ public ImmutableSet<Dialect> acceptDialects() {
     @Override
     public @Nullable String convertSql(String originSql, SessionVariable sessionVariable) {
         String targetURL = GlobalVariable.sqlConverterServiceUrl;
-        if (Strings.isNullOrEmpty(targetURL)) {
+
+        if (Strings.isNullOrEmpty(targetURL) && dialectConverterSerInfo.isEmpty()) {
             return null;
         }
-        return HttpDialectUtils.convertSql(targetURL, originSql, sessionVariable.getSqlDialect());
+
+        if (!Strings.isNullOrEmpty(targetURL)) {
+            String resultStr = HttpDialectUtils.convertSql(targetURL, originSql, sessionVariable.getSqlDialect());
+            if (Strings.isNullOrEmpty(resultStr)) {
+                return originSql;
+            } else {
+                return resultStr;
+            }
+        } else {
+            if (!sessionVariable.isEnableMultiDialectConvertService()) {
+                return originSql;
+            }
+
+            int retryCount = 0;
+            while (retryCount < dialectConverterSerInfo.size()) {
+                String chooseServer = dialectConverterSerInfo.get(chooseServerIdx++ % dialectConverterSerInfo.size());
```

Comment:
We may have to skip the servers that have been marked unavailable, which can be deduced by line 152, for better availability.
e.g. remember some unavailable servers, and reduce the possibility accessing that servers with a back-off algorithm.